### PR TITLE
feat(console,phrases): add support to update sign-in identifiers in user details form

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Support update user sign-in identifiers in user details form.

image

image

## Testing

- [x] Create a new user and modify any information in user details form.
- [x] Delete all user sign-in identifiers including social, then when saving the form a confirm popup shows up.
- [x] Confirm on the popup, user is updated even without any sign-in identifiers
- [x] Cancel on the popup, then the form is still unsaved and you can add one identifier to continue saving.

## Checklist

- [x] `.changeset`

### CODE REVIEW

File: packages/console/src/pages/UserDetails/UserSettings/index.tsx [hunk 1]
```diff
@@ -48,35 +50,42 @@ function UserSettings() {
     if (isSubmitting) {
       return;
     }
-
-    const { customData: inputCustomData, name, avatar } = formData;
+    const { identities, id: userId } = user;
+    const { customData: inputCustomData, username, primaryEmail, primaryPhone } = formData;
+
+    if (!username && !primaryEmail && !primaryPhone && Object.keys(identities).length === 0) {
```
Comment (wangsijie) [inline packages/console/src/pages/UserDetails/UserSettings/index.tsx:56]: nit: can we extract this predicate into a util function?

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/pages/UserDetails/UserSettings/index.tsx (hunk 1 above)
Comment (charIeszhao) [reply packages/console/src/pages/UserDetails/UserSettings/index.tsx:56]: We can, but is there anywhere else also needs this?

File: packages/console/src/pages/Users/components/CreateForm/index.tsx [hunk 2]
```diff
@@ -89,14 +89,18 @@ function CreateForm({ onClose, onCreate }: Props) {
...
-    });
+      setCreatedUserInfo({
+        user: createdUser,
+        password,
+      });
 
-    onCreate();
+      onCreate();
+    } catch {
+      // Do nothing
```
Comment (xiaoyijun) [inline packages/console/src/pages/Users/components/CreateForm/index.tsx:102]: Should update this comment to inform why we do nothing here.

**Review by xiaoyijun — APPROVED** [review]

File: packages/console/src/pages/Users/components/CreateForm/index.tsx (hunk 2 above)
Comment (charIeszhao) [reply packages/console/src/pages/Users/components/CreateForm/index.tsx]: Updated

**Review by gao-sun — APPROVED** [review]
changeset lgtm

**Pull request merged**